packages/main/src/components/Page/index.tsx

```diff
@@ -23,89 +23,101 @@ export interface PagePropTypes extends CommonProps {
-    );
-  };
-
-  private renderTitle = () => <Title level={TitleLevel.H5}>{this.props.title}</Title>;
-
-  private renderHeader = () => {
-    const { showBackButton } = this.props;
-    return (
-      <Bar
-        renderContentLeft={showBackButton ? this.renderBackButton : () => null}
-        renderContentMiddle={this.renderTitle}
-      />
-    );
-  };
-
-  render() {
-    const {
-      children,
-      showFooter,
-      showHeader,
-      classes,
-      className,
-      style,
-      renderCustomHeader,
-      renderCustomFooter,
-      backgroundDesign,
-      tooltip,
-      innerRef,
-      slot
-    } = this.props as PagePropsInternal;
-
-    const pageContainer = StyleClassHelper.of(classes.pageContainer);
-    const headerClasses = StyleClassHelper.of(classes.pageHeader, classes.baseBar);
-    const footerClasses = StyleClassHelper.of(classes.pageFooter, classes.baseBar);
-
-    if (showHeader) {
-      pageContainer.put(classes.pageWithHeader);
-    }
-
-    if (showFooter) {
-      pageContainer.put(classes.pageWithFooter);
-    }
-
-    if (className) {
-      pageContainer.put(className);
-    }
-
-    pageContainer.put(classes[`background${backgroundDesign}`]);
-
-    return (
-      <div ref={innerRef} className={pageContainer.valueOf()} style={style} title={tooltip} slot={slot}>
-        {showHeader && (
-          <header className={headerClasses.valueOf()}>
-            {renderCustomHeader && renderCustomHeader()}
-            {!renderCustomHeader && this.renderHeader()}
-          </header>
-        )}
-        <section className={classes.contentSection}>{children}</section>
-        {showFooter && (
-          <footer className={footerClasses.valueOf()}>{renderCustomFooter && renderCustomFooter()}</footer>
-        )}
-      </div>
-    );
+const useStyles = createUseStyles<string>(styles, {
+  name: 'Page'
+});
+
+const Page = forwardRef((props: PagePropTypes, ref: Ref<HTMLDivElement>) => {
+  const {
+    children,
+    showFooter,
+    showHeader,
+    showBackButton,
+    className,
+    style,
+    renderCustomHeader,
+    renderCustomFooter,
+    backgroundDesign,
+    tooltip,
+    slot,
+    onNavButtonPress,
+    title
+  } = props;
+
+  const classes = useStyles();
+
+  const handleNavBackButtonPress = useCallback(
+    (e) => {
+      if (typeof onNavButtonPress === 'function') {
+        onNavButtonPress(Event.of(null, e.getOriginalEvent()));
+      }
+    },
+    [onNavButtonPress]
+  );
+
+  const renderBackButton = useMemo(
+    () => () => {
+      if (showBackButton) {
+        return (
+          <Button icon="navigation-left-arrow" design={ButtonDesign.Transparent} onClick={handleNavBackButtonPress} />
+        );
+      }
+      return null;
+    },
+    [showBackButton]
+  );
+
+  const renderTitle = useMemo(() => () => <Title level={TitleLevel.H5}>{title}</Title>, [title]);
+
+  const renderHeader = useMemo(
```

Comment:
Whats the point in memoizing a function and calling it below? 
You could just memoize the result of the inner function and also make it dependant on renderHeader and renderCustomHeader.

Reply:
forget the dependant on renderHeader part... just dependant on renderCustomHeader of course